Mathlib/LinearAlgebra/Matrix/Basis.lean

```diff
@@ -165,6 +165,16 @@ def toMatrixEquiv [Fintype ι] (e : Basis ι R M) : (ι → M) ≃ₗ[R] Matrix
       LinearEquiv.apply_symm_apply]
 #align basis.to_matrix_equiv Basis.toMatrixEquiv
 
+variable (R₂) in
+theorem restrictScalars_toMatrix [Fintype ι] [DecidableEq ι] {S : Type*} [CommRing S] [Nontrivial S]
```

Comment:
Do you really need `Nontrivial S` here? It comes from [Basis.restrictScalars](https://leanprover-community.github.io/mathlib4_docs/Mathlib/LinearAlgebra/Basis.html#Basis.restrictScalars), that I think doesn't really need it: if `S` is trivial and `M` is an `S`-module (hence `M = 0)` you can surely produce a basis of any submodule of `M`.

Not sure if it is worth to do it here.

Reply:
Yes, you are right. I'll put that on my TODO list since I am the one who added `Basis.restrictScalars`